src/routes/genres.ts

```diff
@@ -0,0 +1,147 @@
+import { Router, type Response, type Request } from 'express';
+import { Genre } from '../models/genre.model';
+
+const router = Router();
+
+/**
+ * @swagger
+ * /genre:
+ *   get:
+ *     summary: Get a list of genres
+ *     tags:
+ *       - Genres
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *         content:
+ *           application/json:
+ *             example:
+ *               - genreName: Action
+ *               - genreName: Drama
+ */
+router.get('/', async (req: Request, res: Response) => {
+  try {
+    const genres = await Genre.find({});
+    res.status(200).json(genres);
+  } catch (error: any) {
+    res.status(500).json({ message: error.message });
+  }
+});
+
+/**
+ * @swagger
+ * /genre/{id}:
+ *   delete:
+ *     summary: Delete a genre by ID
+ *     tags:
+ *       - Genres
+ *     parameters:
+ *       - in: path
+ *         name: id
+ *         required: true
+ *         description: ID of the genre to delete
+ *         schema:
+ *           type: string
+ *     responses:
+ *       200:
+ *         description: Genre deleted successfully
+ *       404:
+ *         description: Genre not found
+ *       500:
+ *         description: Internal server error
+ */
+router.delete('/:id', async (req: Request, res: Response) => {
```

Comment:
Better to add validation middleware for movies and genres to check field and return 400 (not 500) error in response